packages/ckeditor5-engine/src/model/operation/transform.js

```diff
@@ -86,301 +86,306 @@ function noUpdateTransformation( a ) {
 }
 
 /**
- * Transforms operation `a` by operation `b`.
- *
- * @param {module:engine/model/operation/operation~Operation} a Operation to be transformed.
- * @param {module:engine/model/operation/operation~Operation} b Operation to transform by.
- * @param {module:engine/model/operation/transform~TransformationContext} context Transformation context for this transformation.
- * @returns {Array.<module:engine/model/operation/operation~Operation>} Transformation result.
+ * Operational Transformation main transformation entry point.
  */
-export function transform( a, b, context = {} ) {
-	const transformationFunction = getTransformation( a.constructor, b.constructor );
-
-	/* eslint-disable no-useless-catch */
-	try {
-		a = a.clone();
-
-		return transformationFunction( a, b, context );
-	} catch ( e ) {
-		// @if CK_DEBUG // console.warn( 'Error during operation transformation!', e.message );
-		// @if CK_DEBUG // console.warn( 'Transformed operation', a );
-		// @if CK_DEBUG // console.warn( 'Operation transformed by', b );
-		// @if CK_DEBUG // console.warn( 'context.aIsStrong', context.aIsStrong );
-		// @if CK_DEBUG // console.warn( 'context.aWasUndone', context.aWasUndone );
-		// @if CK_DEBUG // console.warn( 'context.bWasUndone', context.bWasUndone );
-		// @if CK_DEBUG // console.warn( 'context.abRelation', context.abRelation );
-		// @if CK_DEBUG // console.warn( 'context.baRelation', context.baRelation );
-
-		throw e;
-	}
-	/* eslint-enable no-useless-catch */
-}
+export default class OperationTransform {
+	/**
+	 * Transforms operation `a` by operation `b`.
+	 *
+	 * @param {module:engine/model/operation/operation~Operation} a Operation to be transformed.
+	 * @param {module:engine/model/operation/operation~Operation} b Operation to transform by.
+	 * @param {module:engine/model/operation/transform~TransformationContext} context Transformation context for this transformation.
+	 * @returns {Array.<module:engine/model/operation/operation~Operation>} Transformation result.
+	 */
+	static transform( a, b, context = {} ) {
+		const transformationFunction = getTransformation( a.constructor, b.constructor );
+
+		/* eslint-disable no-useless-catch */
+		try {
+			a = a.clone();
+
+			return transformationFunction( a, b, context );
+		} catch ( e ) {
+			// @if CK_DEBUG // console.warn( 'Error during operation transformation!', e.message );
+			// @if CK_DEBUG // console.warn( 'Transformed operation', a );
+			// @if CK_DEBUG // console.warn( 'Operation transformed by', b );
+			// @if CK_DEBUG // console.warn( 'context.aIsStrong', context.aIsStrong );
+			// @if CK_DEBUG // console.warn( 'context.aWasUndone', context.aWasUndone );
+			// @if CK_DEBUG // console.warn( 'context.bWasUndone', context.bWasUndone );
+			// @if CK_DEBUG // console.warn( 'context.abRelation', context.abRelation );
+			// @if CK_DEBUG // console.warn( 'context.baRelation', context.baRelation );
```

Comment:
Can we move this functionality to the new solution as well?

Reply:
And all the other places where we use CK_DEBUG.

Reply:
I think that we could iteratively improve the logger so we could add more functions to it over time (when needed)